Content.Server/StationEvents/Events/GreytideVirusRule.cs

```diff
@@ -89,6 +89,9 @@ protected override void Started(EntityUid uid, GreytideVirusRuleComponent virusC
             if (firelockQuery.HasComp(airlockUid))
                 continue;
 
+            if (HasComp<ExternalAccessMarkerComponent>(airlockUid))
```

Comment:
This should use a query, same as the firelockQuery above

Reply:
Changed it, does it look better now?